operator/scheduler/server_test.go

```diff
@@ -178,103 +178,165 @@ func TestSubscribeServerEvents(t *testing.T) {
 	g := NewGomegaWithT(t)
 
 	type test struct {
-		name               string
-		existing_resources []client.Object
-		results            []*scheduler.ServerStatusResponse
+		name           string
+		existingServer mlopsv1alpha1.Server
```

Comment:
nit: I prefer to keep using the interface client.Object to follow the pattern used in similar tests. What is the reason to have it specific to `mlopsv1alpha1.Server`?

Reply:
No need to cast the server when checking if it has a valid scaling spec.